### PR TITLE
Avoid table names prefixed by an underscore and reserve public

### LINKED ISSUE #6609: Importer fails at renaming table when fed with public.csv
How to reproduce: try to import a CSV file called `public.csv`
Expected: it gets imported.
Actual:
```
2016-02-17 15:25:39 UTC: Before importer run
2016-02-17 15:25:50 UTC: Starting import for /tmp/imports/20160217-17920-1u1xmce/public.csv
2016-02-17 15:25:50 UTC: Unpacking /tmp/imports/20160217-17920-1u1xmce/public.csv
2016-02-17 15:25:51 UTC: Filename: /tmp/imports/20160217-17920-zwrh5u/public.csv Size (bytes): 63677977
...
true: Finished importing data from /tmp/imports/20160217-17920-zwrh5u/public.csv
2016-02-17 15:26:06 UTC: Proceeding to register
2016-02-17 15:26:06 UTC: Before renaming from importer_b30ce374d58a11e59849040111176f01 to public
2016-02-17 15:26:07 UTC: Before moving schema 'public' from cdb_importer to common_data
2016-02-17 15:26:07 UTC: Before persisting metadata 'public' data_import_id: 846e26fc-75d9-4089-a8ad-6c9098bdccca
2016-02-17 15:26:07 UTC: Exception: PG::Error: ERROR:  table "public" does not exist
```

Stacktrace:
```
lib/importer/lib/cartodb-migrator/migrator.rb:102:in `rescue in migrate!'
lib/importer/lib/cartodb-migrator/migrator.rb:107:in `migrate!'
app/models/table.rb:293:in `import_to_cartodb'
app/models/table.rb:376:in `before_create'
app/models/table/user_table.rb:184:in `before_create'
app/models/table.rb:88:in `save'
app/models/table_registrar.rb:18:in `register'
app/connectors/importer.rb:205:in `persist_metadata'
app/connectors/importer.rb:75:in `register'
app/connectors/importer.rb:52:in `block in run'
app/connectors/importer.rb:51:in `each'
app/connectors/importer.rb:51:in `run'
app/models/data_import.rb:638:in `new_importer'
app/models/data_import.rb:372:in `dispatch'
app/models/data_import.rb:172:in `run_import!'
lib/resque/importer_jobs.rb:9:in `block in perform'
lib/resque/base_job.rb:21:in `call'
lib/resque/base_job.rb:21:in `run_action'
lib/resque/importer_jobs.rb:9:in `perform'
```

This shall be easy to fix and might be a cause of stale tables in `cdb_importer` schema (I haven't checked).

Exported tables get that name in many cases (will create a ticket for that in the corresponding repository).

cc/ @zenitraM

### LINKED ISSUE #6622: Table names which start by an underscore (_) should not be supported
Related: https://github.com/CartoDB/cartodb/issues/6609

### PR DESCRIPTION
**Pull request by iriberri**
Fixes #6622 
Fixes #6609 

Through the different pieces of the platform, underscored table names
have been causing issues. (i.e. sequel is crashing when georeferencing a
table whose name starts by an underscore). Although Postgres supports
these names, for each table it creates an internal type preceded by an
underscore. Therefore, there are collisions between underscored names
and already existent types.

This commit will rename any table whose name starts by an underscore.
Besides we have added the word "public" to the array of reserved words,
because it is not officially a reserved word in postgres but it can
cause issues and it is a reserved word in other standards.

### CODE REVIEW

**Comment by iriberri** [comment]
The specs are broken because the user-mover tests are using a table named "public", which we are indeed blocking in this PR. 

Working on that.

cc @zenitraM

**Comment by iriberri** [comment]
Hey @ethervoid, could you please CR? :-)

File: services/user-mover/spec/user_mover_spec.rb [hunk 1]
```diff
@@ -164,9 +164,9 @@ def move_user(user)
   end
 
   def create_tables(user)
-    create_table(user_id: user.id, name: "with_link", privacy: UserTable::PRIVACY_LINK)
-    create_table(user_id: user.id, name: "public",    privacy: UserTable::PRIVACY_PUBLIC)
-    create_table(user_id: user.id, name: "private",   privacy: UserTable::PRIVACY_PRIVATE)
+    create_table(user_id: user.id, name: "with_link_table", privacy: UserTable::PRIVACY_LINK)
```
Comment (ethervoid) [inline services/user-mover/spec/user_mover_spec.rb]: Take this out as constants

File: services/user-mover/spec/user_mover_spec.rb [hunk 2]
```diff
@@ -164,9 +164,9 @@ def move_user(user)
...
+    create_table(user_id: user.id, name: "public_table",    privacy: UserTable::PRIVACY_PUBLIC)
```
Comment (ethervoid) [inline services/user-mover/spec/user_mover_spec.rb]: Use constants here too

**Comment by ethervoid** [comment]
Move to use constants and :+1:

**Comment by iriberri** [comment]
To be :ship: :it: next week, then :-) Thanks!

**Comment by gfiorav** [comment]
retest this, please

**Comment by iriberri** [comment]
@rafatower you told me that you wanted to give this a look too?

**Comment by rafatower** [comment]
:+1: fine by me, the rest are minor improvements

**Comment by iriberri** [comment]
retest this, please

**Comment by luisbosque** [comment]
retest this please

**Pull request merged**